.github/workflows/build-images.yaml

```diff
@@ -30,14 +30,28 @@ jobs:
         id: add-branch-tag
         run: |
           echo "IMG_TAGS=${GITHUB_REF_NAME/\//-} ${{ env.IMG_TAGS }}" >> $GITHUB_ENV
-      - name: Set Authorino version
-        id: authorino-version
+      - name: Set Authorino build info
+        id: authorino-build-info
         run: |
           if [[ ${GITHUB_REF_NAME/\//-} =~ ^v[0-9]+\.[0-9]+\.[0-9]+(-.+)?$ ]]; then
-          echo "VERSION=${GITHUB_REF_NAME/\//-}" >> $GITHUB_ENV
+            tag=${GITHUB_REF_NAME/\//-}
+            echo "version=${tag#v}" >> $GITHUB_ENV
+          elif [[ ${GITHUB_REF_NAME/\//-} == "main" ]]; then
+            echo "version=latest" >> $GITHUB_ENV
           else
-          echo "VERSION=${{ github.sha }}" >> $GITHUB_ENV
+            echo "version=${{ github.ref_name }}" >> $GITHUB_ENV
```

Comment:
It looks like we still have 2 ways to do the same thing more or less. Even though GHA and Makefile both seem to use the git ref, for some reason one does it straightforward and the other uses a file (two files!) in the process.

This GHA workflow (used to build and push the images to registry on pushes to main, releases and for PR testing) uses only the git ref straight away to infer the version info. The Makefile (used locally for dev/testing workflows), on the other hand, reads the version info from `build-recent.yaml`, while still generating this file (from a `build.yaml` "template") before every command, based on the git ref.

I see two paths from here to fix this. We need to decide whether `build.yaml` will be the source for the version info or not.

**OPTION A:** `build.yaml` is the source of version info:

1. The version info should NOT be updated before every command, but simply read from the `build.yaml` file locally (this is the base line, but keep reading to see the special case)
2. Updating `build.yaml` must be a manual step only performed in preparation for releases
3. We need to start doing release branches for Authorino, so these can contain the version of `build.yaml` with the altered version info inside (i.e. "X.Y.Z"), unlike all other branches (including main) whose `build.yaml` will always say "latest"
4. The GHA must read the version info also from `build.yaml`
5. There must be a way for the user to manually override the info in `build.yaml` without modifying the file. This will be the case mostly of when building from feature branches/PRs, both using the Makefile or via GHA. We probably don't need `build-recent.yaml` for that.

**OPTION B:** `build.yaml` is NOT the source:

1. version info must be inferred from the git ref directly, both in the GHA and in the make targets:

| git ref | version info |
|---------|--------------|
| main    | latest       |
| feature | branch-name  |
| vX.Y.Z  | X.Y.Z        |

IMO, OPTION B looks much simpler. On the flip side, it doesn't give us the traceability, along with the code, about which version info would be stamped when building from that source. However, we're also introducing git shas and the `dirty` flag here. So I tend to think that storing the version info along with the code is not needed.

But please let me know what you think @ehearneRedHat @eguzki @didierofrivia

Reply:
@guicassolato I think `OPTION B` looks to be easier but you would lose the traceability as you mentioned, but there would be less files to look to fix. 

I will await other responses before implementing.

Reply:
The `version` info in logs gives high level view of the version being used.  Useful to know if some feature should be there or not. 

For us, developers, `version` is not accurate enough to debug and investigate issues. We want the git sha revision to know exactly the source code revision being run. 

I leave `version` management up to you guys. 

My take? version lives in `version.go`. In `main` branch it is `vX.Y.Z-dev` (where vX.Y.Z has not been released yet). Note that it is not `latest` or `main` as it is meaningless. In release branches it is `vX.Y.Z`. In RELEASE.md file, there is an additional step to update this `version.go` file. This is the approach we are taking in other kuadrant projects.

Reply:
@didierofrivia what are your thoughts ?

Reply:
Going with OPTION B here for simplicity.